dataflow/build.gradle

```diff
@@ -54,46 +55,61 @@ createDataflowShaded('shaded')
+        outputs.file("tests/${dirName}/Test.class")
+
+        delete("tests/${dirName}/Out.txt")
+        delete("tests/${dirName}/Test.class")
+        doLast {
+            javaexec {
+                workingDir = "tests/${dirName}"
+                if (!JavaVersion.current().java9Compatible) {
+                    jvmArgs += "-Xbootclasspath/p:${configurations.javacJar.asPath}".toString()
+                }
+                else if (JavaVersion.current() > JavaVersion.VERSION_11) {
+                    jvmArgs += [
+                            "--add-opens", "jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED",
+                            "--add-opens", "jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED",
+                            "--add-opens", "jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
+                            "--add-opens", "jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED",
+                            "--add-opens", "jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED",
+                            "--add-opens", "jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED",
+                            "--add-opens", "jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED"
+                    ]
+                }
+
+                classpath = sourceSets.test.runtimeClasspath
+                classpath += sourceSets.test.output
+                mainClass = "${pkgName}.${className}"
             }
-            else if (JavaVersion.current() > JavaVersion.VERSION_11) {
-                jvmArgs += [
-                    "--add-opens", "jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED",
-                    "--add-opens", "jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED",
-                    "--add-opens", "jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
-                    "--add-opens", "jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED",
-                    "--add-opens", "jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED",
-                    "--add-opens", "jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED",
-                    "--add-opens", "jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED"
-                ]
+            exec {
+                workingDir = "tests/${dirName}"
+                executable 'diff'
+                args = ['-u', 'Expected.txt', 'Out.txt']
             }
-
-            classpath = sourceSets.test.runtimeClasspath
-            classpath += sourceSets.test.output
-            mainClass = 'livevar.LiveVariable'
-        }
-        exec {
-            workingDir = 'tests/live-variable'
-            executable 'diff'
-            args = ['-u', 'Expected.txt', 'Out.txt']
         }
     }
 }
 
+// run the liveVariableTest
+testDataflowAnalysisDemo("liveVariableTest", "live-variable", "livevar", "LiveVariable")
+// run the busyExpressionTest
+testDataflowAnalysisDemo("busyExpressionTest", "busy-expression", "busyexpr", "BusyExpression")
```

Comment:
I think we don't have "busyExpressionTest" in this PR.

Reply:
Thanks for pointing this out!